pkg/reconciler/revision/background_test.go

```diff
@@ -212,6 +212,65 @@ func TestRateLimit(t *testing.T) {
 	}
 }
 
+func TestRateLimitPerItem(t *testing.T) {
+	logger := logtesting.TestLogger(t)
+
+	var resolves atomic.Int32
+	var resolver resolveFunc = func(ctx context.Context, img string, _ k8schain.Options, _ sets.String) (string, error) {
+		resolves.Inc()
+		return "", errors.New("failed")
+	}
+
+	queue := workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(50*time.Millisecond, 5*time.Second))
+
+	enqueue := make(chan struct{})
+	subject := newBackgroundResolver(logger, resolver, queue, func(types.NamespacedName) {
+		enqueue <- struct{}{}
+	})
+
+	stop := make(chan struct{})
+	done := subject.Start(stop, 1)
+
+	defer func() {
+		close(stop)
+		<-done
+	}()
+
+	start := time.Now()
+	for i := 0; i < 4; i++ {
+		subject.Clear(types.NamespacedName{Name: "rev", Namespace: "ns"})
+		resolution, err := subject.Resolve(rev("rev", "img1", "img2"), k8schain.Options{ServiceAccountName: "san"}, sets.NewString("skip"), 0)
```

Comment:
Should we cache `rev` or at least it's parameters to signify they're the same everywhere?

I ask mainly because I think we're not testing that the backoff in one item isn't impacting another item, right? Should we add a test for that?

Reply:
done, and also added a sub-test that an explicitly different revision isnt limited